lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -103,20 +103,11 @@ class Base
   module Associations
     class Preloader
       prepend(Module.new {
-        def preloaders_for_one(association, records, scope)
-          klass_map = records.compact.group_by(&:class)
-
-          loaders = klass_map.keys.group_by { |klass| klass.virtual_includes(association) }.flat_map do |virtuals, klasses|
-            subset = klasses.flat_map { |klass| klass_map[klass] }
-            preload(subset, virtuals)
-          end
-
-          records_with_association = klass_map.select { |k, _rs| k.reflect_on_association(association) }.flat_map { |_k, rs| rs }
-          if records_with_association.any?
-            loaders.concat(super(association, records_with_association, scope))
-          end
-
-          loaders
+        def preload(records, associations, preload_scope = nil)
+          records = Array.wrap(records).compact
+          associations = records.first.class.replace_virtual_fields(associations) if records.first
```

Comment:
only associations from the first class will be used

can we use `group_by` / `grouped_records` ?

Reply:
also, did polymorphic work before us?
how much of polymorphic did we loose?

Reply:
added grouping by class.